xdsl/dialects/fsm.py

```diff
@@ -0,0 +1,377 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+
+from xdsl.dialects.builtin import (
+    ArrayAttr,
+    ContainerOf,
+    DictionaryAttr,
+    FlatSymbolRefAttr,
+    FunctionType,
+    IndexType,
+    IntegerType,
+    StringAttr,
+)
+from xdsl.ir import (
+    Attribute,
+    Block,
+    Operation,
+    Region,
+    SSAValue,
+)
+from xdsl.ir.core import Dialect
+from xdsl.irdl import (
+    AnyAttr,
+    AnyOf,
+    IRDLOperation,
+    attr_def,
+    irdl_op_definition,
+    operand_def,
+    opt_attr_def,
+    opt_operand_def,
+    region_def,
+    var_operand_def,
+    var_result_def,
+)
+from xdsl.traits import (
+    HasParent,
+    IsTerminator,
+    NoTerminator,
+    SymbolOpInterface,
+    SymbolTable,
+)
+from xdsl.utils.exceptions import VerifyException
+
+signlessIntegerLike = ContainerOf(AnyOf([IntegerType, IndexType]))
+
+
+@irdl_op_definition
+class Transition(IRDLOperation):
+    """Represents a transition of a state with a symbol reference
+    of the next state. This op includes an optional `$guard` region with an `fsm.return`
+    as terminator that returns a Boolean value indicating the guard condition of
+    this transition. This op also includes an optional `$action` region that represents
+    the actions to be executed when this transition is taken"""
```

Comment:
```suggestion
    """
    Represents a transition of a state with a symbol reference
    of the next state. This op includes an optional `$guard` region with an `fsm.return`
    as terminator that returns a Boolean value indicating the guard condition of
    this transition. This op also includes an optional `$action` region that represents
    the actions to be executed when this transition is taken.
    """
```

Please write block comments like this :) (with a newline after `"""`)